### PR TITLE
Add blog post: Building the Neoway's ML Platform with a Team-First Approach and Product Thinking

### PR DESCRIPTION
**Pull request by breno-costa**
@NeowayLabs/ml-platform please review

### CODE REVIEW

File: _posts/2022-01-16-journey-ml-platform.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+### Agile Execution
+
+Agile execution is essential to reduce errors during software development. When we are going to develop some software components for users, we create prototypes to show to a group of users before the development itself. This allows improving before even starting! Software delivery is broken into small parts whenever possible. We generate more value, get feedback and learn from every interaction.
+
+We decided to use [scrum](https://en.wikipedia.org/wiki/Scrum_(software_development)) as a framework for developing and delivering our solutions. Our sprint backlog is composed of development epics previously planned on OKRs, operation, and support tasks. It’s common to have some problem someone faced in the previous week and we need to fix it in the new sprint, or someone needing support to deliver some project.
```
Comment (mfrata) [inline _posts/2022-01-16-journey-ml-platform.md:218]:
```suggestion
We decided to use [scrum](https://en.wikipedia.org/wiki/Scrum_(software_development)) as a framework for developing. Our sprint backlog is composed of development epics previously planned on OKRs, operation, and support tasks. It’s common to have some problem someone faced in the previous week and we need to fix it in the new sprint, or someone needing support to deliver some project. 
```

File: _posts/2022-01-16-journey-ml-platform.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+We decided to use [scrum](https://en.wikipedia.org/wiki/Scrum_(software_development)) as a framework for developing and delivering our solutions. Our sprint backlog is composed of development epics previously planned on OKRs, operation, and support tasks. It’s common to have some problem someone faced in the previous week and we need to fix it in the new sprint, or someone needing support to deliver some project. 
+
+After starting the sprint, we have daily meetings with the entire team to assess progress and resolve any blocks that happen. This meeting is important to keep everyone's commitment to what was agreed. Seeing the progress of the tasks on the board renews the motivation to stay focused and seeing the lack of progress turns on the alert for any problems that might be happening.
```
Comment (mfrata) [inline _posts/2022-01-16-journey-ml-platform.md:220]:
```suggestion
After starting the sprint, we have daily meetings with the entire team to assess progress and resolve any blocks that occurred. This meeting is important to keep everyone's commitment to what was agreed. Seeing the progress of the tasks on the board renews the motivation to stay focused and seeing the lack of progress turns on the alert for any problems that might be happening.
```

File: _posts/2022-01-16-journey-ml-platform.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+Approaching a machine learning platform with product thinking helps to understand the users and generate more value as a lean team
+
+* Understanding your users is the most important thing you have to do. [Do things that don’t scale](http://paulgraham.com/ds.html)!
+* Involve users from the beginning, show them prototypes, let them use the product, get feedback and improve from there. Don’t involve everyone, prefer more mature and curious users with an early adopter profile.
+* A small team needs to optimize for fast and small deliveries. 
+
+Internal teams trust internal platform services if they deliver value, are easy to use, and are reliable.
+
+* You will need the understanding of your users when the services present any problem. Don’t let your users down. Provide good support and you will gain their trust.
+* Use every interaction with your users to get feedback and improve the services. Payment comes in the long term.
```
Comment (mfrata) [inline _posts/2022-01-16-journey-ml-platform.md:244]: Not even a mention to me and @ryukinix ??? 😢

File: _posts/2022-01-19-neoways-ml-platform.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+In the last five years, data science and machine learning have emerged to bring more intelligence in solving customer problems and currently play a key role in the solutions that make up the portfolio. Models, indicators, intelligent journeys are examples of data science-powered applications delivered on our web application or via APIs.
+
+The company's growth brought new challenges to maintain a good communication structure among teams and fast business value delivery to the customers.
+
+# Organizational Structure
+
+For a long time, the company was organized by functional areas like platform, data, data science, application, product. This kind of organization caused functional silos that led to problems such as lack of systemic thinking, and hard dependency management during software development.
```
Comment (ryukinix) [inline _posts/2022-01-19-neoways-ml-platform.md:37]:
```suggestion
For a long time, the company was organized by functional areas like platform, data source, data integration, data science, application, product. This kind of organization caused functional silos that led to problems such as lack of systemic thinking, and hard dependency management during software development.
```

Just data team I think is not very clear for english speakers.

File: _posts/2022-01-19-neoways-ml-platform.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+For instance, the data science team struggled to get historical data in the proper format for the modeling tasks. The team also had difficulties deploying model predictions integrated with the company's architecture. Both tasks had dependencies with internal teams and the services were not provided properly.
+
+That organizational structure resulted in inefficiency to deliver business value to our customers. At the end of 2019, the company was redesigned into cross-functional teams focused on business domains capable of delivering end-to-end solutions and accelerating software delivery.
```
Comment (ryukinix) [inline _posts/2022-01-19-neoways-ml-platform.md:41]:
```suggestion
That organizational structure resulted in inefficiency to deliver business value to our customers. At start of 2020, the company was redesigned into cross-functional teams focused on business domains capable of delivering end-to-end solutions and accelerating software delivery.
```

I remember this differently. If I recall correctly, the team redesign was in the beginning of 2020, no? :thinking: 

@mfrata any opinion here?

File: _posts/2022-01-19-neoways-ml-platform.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+The image below presents a very summarized view of some cells with which our machine learning platform team interacts the most.
+
+<figure>
+  <img src="/images/posts/neoways-ml-platform/cells_overview.png" alt="">
+</figure>
+
+The product and service cells are responsible for delivering value directly to the end customers. They need to be empowered to build and deliver customer value as quickly and independently as possible. Those cells are created to better serve the business domain requirements.
+
+On the other hand, the platform cells have two responsibilities. The first one is to simplify the use of complex technologies that require a high level of specialization, such as Kubernetes, Spark clusters, distributed systems, data lake, streaming platform, IAM, security, data warehouses, search engines. The list goes on.
```
Comment (ryukinix) [inline _posts/2022-01-19-neoways-ml-platform.md:67]:
```suggestion
On the other hand, the platform cells have two responsibilities. The first one is to simplify the use of complex technologies that require a high level of specialization, such as Kubernetes, Spark clusters, distributed systems, data lake, streaming platform, IAM, security, data warehouses, search engines... The list goes on.
```

I think this punctuation style (ellipsis) makes more sense for a long sequence of list than a period (maybe without end).

File: _posts/2022-01-19-neoways-ml-platform.md [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+
+A technical team may be tempted to solve the problems only from a technical perspective by considering aspects such as scalability, reliability, and maintainability. Those things are important, but the platform solutions can’t fail at a crucial point: user adoption.
+
+In the blog post [Mind the platform execution gap](https://martinfowler.com/articles/platform-prerequisites.html), the authors have suggested prioritizing developer experience, and it’s one of the aspects to mind the platform execution gap. The reason is simple: a product that no one uses is not a successful product, no matter its technical merits.
+
+Our team decided to approach the platform with product thinking. We decided to focus on the user needs and experience, but deliver solutions with technical excellence. 
+
+From the beginning, we interviewed the users, tried to understand their journey, and listened to their pain points. It's necessary to see any user perspective on the platform components being developed. An advantage of internal products is that we have closer contact with users through the company’s communication channels. **User empathy is the key!**
```
Comment (ryukinix) [inline _posts/2022-01-19-neoways-ml-platform.md:155]: I cannot agree more with this!

**Review by ryukinix — APPROVED** [review]
I think the overall article is great. I added just a few minor suggestions to the text.

@breno-costa congratulations!

**Review by mfrata — APPROVED** [review]

**Pull request merged**